Src/Eiffel/eiffel/byte_code/visitor/assertion_breakable_slot_strategy.e

```diff
@@ -0,0 +1,481 @@
+		end
+
+	process_if_b (a_node: IF_B)
+		do
+			if attached {HIDDEN_IF_B} a_node then
+				Precursor (a_node)
+			else
+					-- Generate hook for the condition test
+				increment_counter (a_node)
+
+				Precursor (a_node)
+			end
+		end
+
+	process_if_expression_b (a_node: IF_EXPRESSION_B)
+			-- <Precursor>
+
+		do
+			a_node.condition.process (Current)
+
+				-- Generate hook for Then_part.
+			increment_counter (a_node)
+
+				-- Generate expression for Then_part.
+			a_node.then_expression.process (Current)
+
+			if attached a_node.elsif_list as l then
+					-- Generate byte code for alternatives.
+				across
+					l as c
+				loop
+					c.item.process (Current)
+				end
+			end
+
+				-- Generate hook for Else_part.
+			increment_counter (a_node)
+
+				-- Generate expression for Else_part.
+			a_node.else_expression.process (Current)
+		end
+
+	process_inspect_b (a_node: INSPECT_B)
+			-- Process `a_node'.
+		do
+			increment_counter (a_node)
+
+				-- Generate switch expression byte code
+			a_node.switch.process (Current)
+			if a_node.case_list /= Void then
+				across
+					a_node.case_list as c
+				loop
+					c.item.process (Current)
+				end
+					-- Generate code for the various inspect matches
+				a_node.case_list.process (Current)
+			end
+			if a_node.else_part /= Void then
+					-- We need to pop the value of the expression since we do not need it anymore.
+				a_node.else_part.process (Current)
+			end
+		end
+
+	process_instr_call_b (a_node: INSTR_CALL_B)
+			-- Process `a_node'.
+		do
+			increment_counter (a_node)
+			Precursor (a_node)
+		end
+
+	process_loop_b (a_node: LOOP_B)
+			-- Process `a_node'.
+		local
+			invariant_breakpoint_slot: INTEGER
+			body_breakpoint_slot: INTEGER
+			l_context: like context
+			l_old_hidden_code_level: INTEGER
+		do
+			l_context := context
+			l_old_hidden_code_level := l_context.hidden_code_level
+			l_context.set_hidden_code_level (0)
+			if attached a_node.iteration_initialization as i then
+					-- Generate byte code for iteration initialization.
+				increment_counter (a_node)
+				l_context.enter_hidden_code
+				i.process (Current)
+				l_context.exit_hidden_code
+			end
+			if a_node.from_part /= Void then
+					-- Generate byte code for the from part
+				a_node.from_part.process (Current)
+			end
+
+			if a_node.variant_part /= Void then
+					-- ...
+			end
+
+				-- Record context.
+			invariant_breakpoint_slot := l_context.get_breakpoint_slot
+
+			if not (a_node.invariant_part = Void and then a_node.variant_part = Void) then
+					-- Invariant loop byte code
+				if a_node.invariant_part /= Void then
+					l_context.set_assertion_type ({ASSERT_TYPE}.in_loop_invariant)
+					a_node.invariant_part.process (Current)
+				end
+					-- Variant loop byte code
+				if a_node.variant_part /= Void then
+					l_context.set_assertion_type ({ASSERT_TYPE}.in_loop_variant)
+					a_node.variant_part.process (Current)
+				end
+				l_context.set_assertion_type (0)
```

Comment:
How does it work if a loop expression is used inside an assertion?

Reply:
Can we have a standard loop in assertion?
a LOOP_EXPR_B ..., yes, but not a LOOP_B , isn't it?

Reply:
but I guess comment also apply to LOOP_EXPR_B

Reply:
Yes, the comment is for both.

Also, I'm not sure about inline agents, how they are processed.